### ISSUE TITLE
UpdateReplacePolicy is removed from AWS::S3::Bucket on transform

### ISSUE THREAD

**Issue by mhart** [aws/serverless-application-model#1807]
**Description:**

`cfn-lint` highlighted an issue where SAM removes `UpdateReplacePolicy` from an `AWS::S3::Bucket` resource: https://github.com/aws-cloudformation/cfn-python-lint/issues/1265#issuecomment-695314794

I commented over at https://github.com/aws/serverless-application-model/issues/450#issuecomment-695323146 but there has been no response in a few months (it's actually unclear whether this is the same issue).

**Steps to reproduce the issue:**

Original SAM template:

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:

  Bucket:
    Type: AWS::S3::Bucket
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain

  Function:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs12.x
      Events:
        ObjectCreated:
          Type: S3
          Properties:
            Bucket: !Ref Bucket
            Events: s3:ObjectCreated:*
```

**Observed result:**

(just the bucket portion)

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  Bucket:
    DeletionPolicy: Retain
    DependsOn:
    - FunctionObjectCreatedPermission
    Properties:
      NotificationConfiguration:
        LambdaConfigurations:
        - Event: s3:ObjectCreated:*
          Function:
            Fn::GetAtt:
            - Function
            - Arn
    Type: AWS::S3::Bucket
```

**Expected result:**

(just the bucket portion)

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  Bucket:
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain
    DependsOn:
    - FunctionObjectCreatedPermission
    Properties:
      NotificationConfiguration:
        LambdaConfigurations:
        - Event: s3:ObjectCreated:*
          Function:
            Fn::GetAtt:
            - Function
            - Arn
    Type: AWS::S3::Bucket
```

**Comment by sriram-mv** [comment]
Got a repro and its a fairly simple fix.

**Comment by mndeveci** [comment]
Closing this issue since it is been released with [SAM v1.36.0](https://github.com/aws/serverless-application-model/releases/tag/v1.36.0) and [SAM CLI v1.24.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.24.0).

You can read more about how resource level attributes are handled in SAM via our updated documentation; https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification-resource-attributes.html

Thanks!
